Add APIRule 404 troubleshooting guide (#10909)
kyma-project/kyma: docs/api-gateway/10-02-connect-api-rule.md

  ```

  >**TIP:** Name of the VirtualService consists of the name of the APIRule and a random suffix.

> Sometimes Oathkeeper Maester controller stops reconciling Rules on long living clusters. This can result in random `404 Not Found` responses, because Oathkeeper does not contain Rules reflecting the actual state of the cluster. A simple restart of the Pod resolves the issue, but you might want to verify if that is the issue you have encountered.

Reviewer: ```suggestion
Sometimes Oathkeeper Maester controller stops reconciling Rules on long-living clusters. This can result in random `404 Not Found` responses, because Oathkeeper does not contain Rules reflecting the actual state of the cluster. A simple restart of the Pod resolves the issue, but you might want to verify if that is the issue you have encountered.
```

1. Fetch all Oathkeeper Pods' names:

    ```bash
    kubectl get pods -n kyma-system -l app.kubernetes.io/name=oathkeeper -o jsonpath='{.items[*].metadata.name}'
    ```

2. Fetch the access Rules from every Oathkeeper Pod and save them to a file:

    ```bash
   kubectl cp -n kyma-system -c oathkeeper "{POD_NAME}":etc/rules/access-rules.json "access-rules.{POD_NAME}.json" 
   ```

> 3. If you have more than one instance of Oathkeeper, compare whether the files contain the same Rules. Othkeeper stores Rules as a JSON, so you might want to use [jd](https://github.com/josephburnett/jd) to automate it:

Reviewer 2: ```suggestion
3. If you have more than one instance of Oathkeeper, compare whether the files contain the same Rules. Oathkeeper stores Rules as a JSON, so you might want to use [jd](https://github.com/josephburnett/jd) to automate it:
```

> 3. If you have more than one instance of Oathkeeper, compare whether the files contain the same Rules. Oathkeeper stores Rules as a JSON, so you might want to use [jd](https://github.com/josephburnett/jd) to automate it:

Reviewer: ```suggestion
3. If you have more than one instance of Oathkeeper, compare whether the files contain the same Rules. Oathkeeper stores Rules as a JSON, so you might want to use [jd](https://github.com/josephburnett/jd) to automate the comparison:
```

    ```bash
   jd -set {FIRST_FILE} {SECOND_FILE} 
   ```

>     Oathkeeper Pods are out of sync if there are any differences between the files (except the order of Rules).

Reviewer 3: ```suggestion
    Oathkeeper Pods are out of sync if there are any differences between the files other than the order of Rules.
```
